docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
This is something I'd love to have open discussion about. My vision for features section is that it will be *the* user specified configuration, while domain (VM Spec) section is more of a runtime, low level description. If we generate VM Spec from features, there should be no conflict. If user directly modifies VM Spec, it should be considered power user task that may lead to inconsistencies (think of it as hooking mechanism in VDSM).